src/dymaptic.GeoBlazor.Core/Components/Views/MapView.razor.cs

```diff
@@ -2249,80 +2112,204 @@ protected override async Task RenderView(bool forceRender = false)
+                await RemoveWidget(widget);
+
+                break;
+            case Graphic graphic:
+                await RemoveGraphic(graphic);
+
+                break;
+            case Constraints _:
+                Constraints = null;
+
+                break;
+            case SpatialReference _:
+                SpatialReference = null;
+
+                break;
+            case Extent _:
+                Extent = null;
+
+                break;
+            case HighlightOptions _:
+                HighlightOptions = null;
+
+                break;
+            default:
+                await base.UnregisterChildComponent(child);
+
+                break;
         }
     }
 
-    /// <summary>
-    ///     Removes a widget from the view.
-    /// </summary>
-    public async Task RemoveWidget(Widget widget)
+    /// <inheritdoc />
+    internal override void ValidateRequiredChildren()
     {
-        if (ViewJsModule is null) return;
+        base.ValidateRequiredChildren();
+        Map?.ValidateRequiredChildren();
+        Constraints?.ValidateRequiredChildren();
+        SpatialReference?.ValidateRequiredChildren();
+        Extent?.ValidateRequiredChildren();
 
-        if (_widgets.Contains(widget))
+        foreach (Graphic graphic in Graphics)
         {
-            _widgets.Remove(widget);
-            widget.Parent = null;
+            graphic.ValidateRequiredChildren();
         }
 
-        await InvokeAsync(async () =>
+        foreach (Widget widget in Widgets)
+        {
+            widget.ValidateRequiredChildren();
+        }
+    }
+    
+    private async Task<HitTestResult> HitTestImplementation(object pointObject, HitTestOptions? options,
+        bool isEvent)
+    {
+        try
+        {
+            if (IsServer)
+            {
+                var eventId = Guid.NewGuid();
+
+                await ViewJsModule!.InvokeVoidAsync("hitTest", CancellationTokenSource.Token,
+                    pointObject, eventId, Id, isEvent, options);
+                var json = _hitTestResults[eventId].ToString();
+                _hitTestResults.Remove(eventId);
+
+                return JsonSerializer.Deserialize<HitTestResult>(json,
+                    new JsonSerializerOptions { PropertyNameCaseInsensitive = true })!;
+            }
+
+            return await ViewJsModule!.InvokeAsync<HitTestResult>("hitTest",
+                CancellationTokenSource.Token, pointObject, null, Id, isEvent, options);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex);
+        }
+
+        return new HitTestResult(new ViewHit[] { }, new ScreenPoint(1, 1));
+    }
+    
+    private bool IsPro()
+    {
+        if (_isPro is null)
         {
             try
             {
-                await ViewJsModule!.InvokeVoidAsync("removeWidget",
-                    CancellationTokenSource.Token, widget.Id, Id);
+                Assembly _ = Assembly.Load("dymaptic.GeoBlazor.Pro");
```

Comment:
is there a better way to check then throwing an exception?

Reply:
I can't find it. There is `AppDomain.CurrentDomain.GetAssemblies()` which you can iterate through, but that says it gets "loaded" assemblies, and I'm not sure if it would be loaded at this point before calling `Assembly.Load`.